packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -97,6 +103,24 @@ Learn more with our localization guide. https://sfdc.co/localization-guide
     // Get the current currency to be used through out the app
     const currency = locale.preferredCurrency || l10n.defaultCurrency
 
+    // Handle creating a new pasket if there isn't one already assigned to the current
+    // customer.
+    const {data: customer} = useCurrentCustomer()
+    const {data: baskets} = useCustomerBaskets(
+        {parameters: {customerId: customer.customerId}},
+        {enabled: !!customer.customerId && !isServer}
+    )
+    const createBasket = useShopperBasketsMutation('createBasket')
+
+    // Create a new basket if the current customer doesn't have one.
+    useEffect(() => {
+        if (baskets?.total === 0) {
+            createBasket.mutate({
+                body: {}
+            })
+        }
+    }, [baskets])
```

Comment:
`useEffect` dependencies are compared by `==` (object references), let's use primitive types instead of array/object when possible.

```suggestion
    }, [baskets?.total])
```